### DISCUSSION TITLE
0.90.x Discussion

### DISCUSSION THREAD

**Discussion by wingkwong** [nocodb/nocodb#1897] [General]
In case you've any thoughts about the new design or any doubts related to the migrations.

**Comment by RawmanGrey** [comment]
First things first - thank you all for such a dedicated work! Recent weeks where huge 🙏

I have one concern about API design, briefly touched in #1898, - using names(columns, tables and projects) in endpoints seems wrong. I'm not much experienced dev, but I think it's a better practice to use IDs. Apart from from avoiding collisions, such decision will benefit internationalization (user friendly localized names excessively load network) and provide more solid ground for third party apps.

**Reply by o1lab** [reply]
@RawmanGrey - thanks for all the feedback you have been giving. I believe, you can use projectid/tableid interchangeably in place of names - its not documented yet.

**Reply by RawmanGrey** [reply]
Yeah, but most of the pain is in column numbers and there are even two existing properties for any column - "title" and "column_name" already : ) It would be great to have a similar to the form view option to tune user-facing names in all views one day 🙄
